### DISCUSSION TITLE
GFX11 MIOpen Support?

### DISCUSSION THREAD

**Discussion by Catba11** [ROCmSoftwarePlatform/MIOpen#1904] [Q&A] [answered]
Dear MIOpen developers,

I bought a gfx1100 card and just realized there's no current MIOpen support for GFX11/RDNA3 GPUs. Before selling the card I got, I'm wondering if there's any plan or roadmap to support GFX11 cards? 

And if possible, is there any workaround to run PyTorch for now?

Thank you!

**Comment by xuhuisheng** [comment]
i will suggest traditional workaround `export HSA_OVERRIDE_GFX_VERSION=10.3.0`, please let us know if it works. thank you.

**Reply by atamazov** [reply]
> i will suggest traditional workaround `export HSA_OVERRIDE_GFX_VERSION=10.3.0`, please let us know if it works. thank you.

Does this correspond to gfx1030?

**Reply by jellersby** [reply]
> i will suggest traditional workaround `export HSA_OVERRIDE_GFX_VERSION=10.3.0`, please let us know if it works. thank you.

Unfortunately it doesn't work for me. The function that always gives me trouble is hipMemcpy. I've tried it in a standalone C++ program, as well as via Pytorch. Here's an example gdb stacktrace from Pytorch (ROCm 5.2):

```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.

#0  0x00007fff8a31bc92 in ihipMemcpyCommand(amd::Command*&, void*, void const*, unsigned long, hipMemcpyKind, amd::HostQueue&) ()
   from /home/jellersby/Development/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/lib/libamdhip64.so
#1  0x00007fff8a31c443 in ihipMemcpy(void*, void const*, unsigned long, hipMemcpyKind, amd::HostQueue&, bool) () from /home/jellersby/Development/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/lib/libamdhip64.so
#2  0x00007fff8a3566d1 in hipMemcpyWithStream () from /home/jellersby/Development/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/lib/libamdhip64.so

<rest of stacktrace discarded>
```

**Comment by atamazov** [comment]
@fruitpudding Currently ROCm does not support gfx11 HW, please refer to https://docs.amd.com/bundle/Hardware_and_Software_Reference_Guide/page/Hardware_and_Software_Support.html. We can consider supporting gfx11 in MIOpen only after ROCm supports it.

However, you can try the workaround recommended by @xuhuisheng + more MIOpen specific workarounds (TBD). Stay tuned.

**Comment by atamazov** [comment]
@fruitpudding FYI there is a ticket dedicated to gfx11 support in ROCm, https://github.com/RadeonOpenCompute/ROCm/issues/1813

**Comment by littlewu2508** [comment]
There are gfx11 enabled in rocBLAS, so I suppose there are ongoing progress for ROCm math libraries, but it's not completed yet

**Comment by jellersby** [comment]
It looks like there is some movement on gfx11 support in miopen: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1925

**Reply by jellersby** [reply]
Can we expect gfx11 support in ROCm 5.5? :innocent: :pray:

**Reply by atamazov** [reply]
MIOpen already has the most of necessary support for it. About ROCm 5.5, please ask here: https://github.com/RadeonOpenCompute/ROCm/issues/1813

**Comment by PennyFranklin** [comment]
months later and it seems still do not support RDNA3 yet

**Reply by atamazov** [reply, marked as answer]
you can take the most recent `develop` branch and build it. gfx11 support is there.

**Reply by PennyFranklin** [reply]
can you please give a more detailed guidance?I tride but failed to find the package match my gfx1100.

**Comment by huanrwan-amd** [comment]
@fruitpudding @PennyFranklin GFX11/RDNA3 GPUs are supported in the latest ROCm release. Please let us know if your need more help.

**Reply by PennyFranklin** [reply]
Already feeling the effort from the developers. Prior to ROCm6.3, the quick installation guide in ROCm Documentation was not compatible with Ubuntu at all. There were two things that would happen: an error would be reported after the installation, and then a bunch of errors would occur when` sudo apt update` was used on the terminal. Second, after installing ROCm and dkms driver, edge browser and VSCode application cannot be opened, but both can be opened normally once the driver is uninstalled. Immediately after ROCm6.3 came out, I re-installed Ubuntu24.04 and used the quick installation guide in ROCm Documentation. The installation was smooth and compatible with edge as well as VSCode and anaconda. Now I'm waiting for ROCm for Windows, after all, I use PC for playing more , and I installed Linux and ROCm because it's more efficient than torch-dml, faster to run thesis projects and draw pictures on sdwebui. I bet you don't want to switch systems back and forth every time something happens.
